### ISSUE TITLE
Regress in 3.12.x breaks links to headers that themselves contain links

### ISSUE THREAD

**Issue by soul-codes** [tcort/markdown-link-check#348]
Test case

```md
This is a [link to a linked header](#this-header-is-linked)

# This header is [linked](#somewhere)

## Somewhere
```

This reports a deadlink for `#this-header-is-linked` for 3.12.x but a live link for 3.11.2.

Thanks for this wonderful tool! 🙏 Happy to dig and PR if desired.

**Comment by per1234** [comment]
This was fixed by b2beaa0 (https://github.com/tcort/markdown-link-check/pull/328).

```text
$ printf 'This is a [link to a linked header](#this-header-is-linked)\n\n# This header is [linked](#somewhere)\n\n## Somewhere' > /tmp/foo.md

$ npx github:tcort/markdown-link-check#3c31a25 /tmp/foo.md  # Last bad commit.

FILE: C:/Users/per/AppData/Local/Temp/foo.md
  [✖] #this-header-is-linked
  [✓] #somewhere

  2 links checked.

  ERROR: 1 dead links found!
  [✖] #this-header-is-linked → Status: 404

$ npx github:tcort/markdown-link-check#b2beaa0 /tmp/foo.md  # First good commit.

FILE: C:/Users/per/AppData/Local/Temp/foo.md
  [✓] #this-header-is-linked
  [✓] #somewhere

  2 links checked.
```
